src/main/java/org/jenkins/tools/test/hook/JacocoHook.java

```diff
@@ -18,7 +18,9 @@ public class JacocoHook extends PluginCompatTesterHookBeforeExecution {
     @Override
     public boolean check(@NonNull BeforeExecutionContext context) {
         Model model = context.getModel();
-        return "jacoco".equals(model.getArtifactId());
+        return "org.jenkins-ci.plugins".equals(model.getGroupId())
+                && "jacoco".equals(model.getArtifactId())
+                && "hpi".equals(model.getPackaging());
```

Comment:
Elsewhere we check group ID, artifact ID, and packaging, so do so here as well.

Reply:
elsewhere we are going the wrong way - we should just check the `pluginID` as we are in the business of checking plugins not artifacts, but 🤷 
aka  `context.getPlugin().name.equals("jacoco");`

Reply:
I guess we just have different philosophies. I prefer to be as paranoid as possible, possibly from my days doing kernel development.